compiler/parser/chapel.ypp

```diff
@@ -542,37 +542,41 @@ stmt:
   deprecated_decl_stmt
 | include_module_stmt
 | block_stmt
+| executable_stmt
 | use_stmt
 | import_stmt
 | require_stmt
-| assignment_stmt
 | extern_block_stmt
-| if_stmt
 | implements_stmt
 | interface_stmt
-| loop_stmt
-| select_stmt
 | defer_stmt
 | try_stmt
 | throw_stmt
 | return_stmt
+| TATOMIC stmt                         { $$ = buildAtomicStmt($2); }
+| TBREAK opt_label_ident TSEMI         { $$ = buildGotoStmt(GOTO_BREAK, $2); }
+| TCONTINUE opt_label_ident TSEMI      { $$ = buildGotoStmt(GOTO_CONTINUE, $2); }
+| TLABEL ident_def stmt                { $$ = buildLabelStmt($2, $3); }
+| TYIELD expr TSEMI                    { $$ = buildPrimitiveStmt(PRIM_YIELD, $2); }
+| error TSEMI                          { $$ = buildErrorStandin(); }
+;
+
+executable_stmt:
```

Comment:
Could we move `throw_stmt` to executable statement as well? It is pretty silly, but sometime in the past I've found myself trying to write `try! throw new Error();` when placing placeholders for things that will later throw.

Reply:
Did this in my latest commit, and did some other related cleanups if you want to take another look.

Reply:
Oh, and I also opened issue #18938 to more generally discuss what statements `try`/`try!` should or should not apply to since the spec is vague on it and in case anything I did here causes concern going forward.  I don't think this PR needs to wait for this because the status quo was so broken compared to what it was intended to do (as far as I've always imagined anyway?), so I'm thinking of that issue as a retrospective design discussion of sorts.